src/scheduler.cc

```diff
@@ -3,16 +3,138 @@
 #include <random>
 #include <thread>
 #include <chrono>
+#include <sstream>
 
 #include "utils.h"
 
+#ifndef NDEBUG
+template<>
+class SchedulerService::MySynchronizedPtr<void> {
+  SchedulerService* me_;
+  size_t order_;
+  unsigned int line_number_;
+  static unsigned long long get_thread_id() {
+    unsigned long long id = 0;
+    std::stringstream ss;
+    ss << std::this_thread::get_id();
+    ss >> id;
+    // This seems to fail on a Mac. I unfortunately can't find a workaround.
```

Comment:
Can you elaborate on this comment? What fails? And what is the failure (e.g., does it crash? or does the check not actually check anything?)